sdk/azcore/response_test.go

```diff
@@ -7,6 +7,7 @@ package azcore
 
```

Comment:
I am curious that should we unify the package name of the test packages? I am seeing one of the test package `azcore_test` here: https://github.com/Azure/azure-sdk-for-go/blob/master/sdk/azcore/example_test.go

Reply:
Good point! Makes sense to make these also azcore_test. I can update this in a separate PR though.

Reply:
So I believe the name `azcore_test` is intentional in this case.  When you use the package name `<package>_test` it means you don't have access to unexported content within `<package>`.  This is useful when writing examples for two reasons.

- the examples will have to use the fully qualified name, e.g. `azcore.Poller` which is nice for copy/paste
- it ensures you don't have access to unexported content

Reply:
Gotcha, will leave as is.